src/TRestTrackAlphaAnalysisProcess.cxx

```diff
@@ -0,0 +1,281 @@
+     fOutTrackEvent->SetNumberOfYTracks(1);
+     volHit.RemoveHits();
+     delete trackYZ;
+
+
+     if ( std::abs(smtTckX.front().z - originZ) >= std::abs(smtTckX.back().z - originZ)  ){
+       originX = smtTckX.back().xy;
+       endX = smtTckX.front().xy;
+     } else {
+       endX = smtTckX.back().xy;
+       originX = smtTckX.front().xy;
+     }
+
+     if ( std::abs(smtTckY.front().z - originZ) >= std::abs(smtTckY.back().z - originZ)  ){
+       originY = smtTckY.back().xy;
+       endY = smtTckY.front().xy;
+     } else {
+       endY = smtTckY.back().xy;
+       originY = smtTckY.front().xy;
+     }
+
+    debug<<"Origin x: "<<originX<<" y: "<<originY<<" z: "<<originZ<<endl;
+    debug<<"End x: "<<endX<<" y: "<<endY<<" z: "<<endZ<<endl;
+
+    double dX = (originX - endX);
+    double dY = (originY - endY);
+    double dZ = (originZ - endZ);
+    Double_t length = TMath::Sqrt(dX*dX+dY*dY+dZ*dZ);
+    Double_t angle = TMath::ACos(dZ/length);
+
+    debug<<"Track length "<<length<<" angle: "<<angle<<endl;
+
+    // A new value for each observable is added
+    SetObservableValue("originX", originX);
+    SetObservableValue("originY", originY);
+    SetObservableValue("originZ", originZ);
+    SetObservableValue("endX", endX);
+    SetObservableValue("endY", endY);
+    SetObservableValue("endZ", endZ);
+    SetObservableValue("length", length);
+    SetObservableValue("angle", angle);
+    SetObservableValue("downwards", downwards);
+    SetObservableValue("totalEnergy", trackEnergy);
+
+    return fOutTrackEvent;
+}
+
+std::vector<alphaTrackHit> TRestTrackAlphaAnalysisProcess::smoothTrack(std::vector <alphaTrackHit> &hits){
+
+  std::vector <alphaTrackHit> res;
+  std::sort(hits.begin(),hits.end(),alphaTrackHit::sortByTime);
+  int i=0;
+  const int size = hits.size();
+  debug<<"Track size "<<size<<endl;
+    while(i<size){
+      double posXY=0;
+      double en=0;
+      alphaTrackHit hit = hits[i];
+      double posZ = hits[i].z;
+      int nPoints=0;
+      while(posZ == hits[i].z && i<size){
+        posXY += hits[i].xy*hits[i].en;
+        en += hits[i].en;
+        i++;
+        nPoints++;
+      }
+        if(en>0 && nPoints >1){
+          hit.xy = posXY/en;
+          hit.en = en;
+          res.emplace_back(hit);
+        }
+
+    }
+
+  debug<<"Res size "<<res.size()<<endl;
+
+  return res;
+}
+
+//______________________________________________________________________________
+void TRestTrackAlphaAnalysisProcess::EndProcess() {
+    // Function to be executed once at the end of the process
+    // (after all events have been processed)
+
+    // Start by calling the EndProcess function of the abstract class.
+    // Comment this if you don't want it.
+    // TRestEventProcess::EndProcess();
+}
+
+//______________________________________________________________________________
+void TRestTrackAlphaAnalysisProcess::InitFromConfigFile() {
+  fTrackBalance = StringToDouble(GetParameter("trackBalance", "0.65"));
```

Comment:
There is no need to implement this method. Simply remove completely `InitFromConfigFile` from `TRestTrackAlphaAnalysisProcess`.

Then, `TRestEventProcess::InitFromConfigFile` will be responsible to identify a `fDummyVariable` with the parameter given in the config file `dummyVariable`.

In the case of `trackBalance` you just need that a data member named `fTrackBalance` exists.

The declaratio at the header would look as:

```
/// A parameter to define the track balance ... bla bla bla
Double_t fTrackBalance = 0.65;
```

Reply:
Actually the methods `LoadDefaultConfig()`, `LoadConfig()`, `TRestTrackAlphaAnalysisProcess(char* cfgfile)` are not needed either.